### PR TITLE
bump operator-frawork-api

### PR DESCRIPTION
**Pull request by camilamacedo86**
Description
bump operator-frawork-api to get its latest changes and fixes (https://github.com/operator-framework/api/compare/v0.10.7...f579141ce61d8bb7c4dc7c06169bc436ac4b9703)

### CODE REVIEW

File: changelog/fragments/bump-operator-frameworkapi.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      Optional `OperatorHub.io` validator called by `operator-sdk bundle validate ./bundle --select-optional name=operatorhub` now allow the missing category `Modernization & Migration`.
```
Comment (asmacdo) [inline changelog/fragments/bump-operator-frameworkapi.yaml:5]:
```suggestion
      Optional `OperatorHub.io` validator called by `operator-sdk bundle validate ./bundle --select-optional name=operatorhub` now includes a previously missing category `Modernization & Migration`.
```

File: changelog/fragments/bump-operator-frameworkapi.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    kind: "bugfix"
+    breaking: false
+  - description: >
+      Fixes error message faced to validate bundle spec and provide a better understatement when invalid sevice accounts are found.
```
Comment (asmacdo) [inline changelog/fragments/bump-operator-frameworkapi.yaml:9]:
```suggestion
      Improves the validate bundle spec error message top rovide a better explanation when invalid service accounts are found.
```

**Review by asmacdo — CHANGES_REQUESTED** [review]

File: changelog/fragments/bump-operator-frameworkapi.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# entries is a list of entries to include in
+# release notes and/or the migration guide
+entries:
+  - description: >
+      Optional `OperatorHub.io` validator called by `operator-sdk bundle validate ./bundle --select-optional name=operatorhub` now includes a previously missing category `Modernization & Migration`.
+    kind: "bugfix"
+    breaking: false
+  - description: >
+      Improves the validate bundle spec error message top rovide a better explanation when invalid service accounts are found.
```
Comment (asmacdo) [inline changelog/fragments/bump-operator-frameworkapi.yaml:9]:
```suggestion
      Improves the validate bundle spec error message to provide a better explanation when invalid service accounts are found.
```

Comment (asmacdo) [reply changelog/fragments/bump-operator-frameworkapi.yaml:9]: This ones my fault :sob:

**Review by asmacdo — APPROVED** [review]
/lgtm

**Comment by asmacdo** [comment]
@camilamacedo86 you might have a hard time finding another LGTM this week. Since this is pretty trivial, I'm comfortable if you want to merge when CI passes.

**Review by rashmigottipati — APPROVED** [review]
/lgtm

**Pull request merged**